File: backend/java/src/demo/src/main/java/com/example/demo/course/entity/Course.java
package com.example.demo.course.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "courses")
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "teacher_id", nullable = false)
    private Long teacherId;

    @Column(name = "school_id", nullable = false)
    private Long schoolId;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "cover_image")
    private String coverImage;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CourseStatus status = CourseStatus.PRE_RELEASE;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CourseVisibility visibility = CourseVisibility.PUBLIC;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CoursePermission permission = CoursePermission.OPEN;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    public enum CourseStatus {
        PRE_RELEASE, IN_PROGRESS, COMPLETED, HIDDEN
    }

    public enum CourseVisibility {
        PUBLIC, RESTRICTED, PRIVATE
    }

    public enum CoursePermission {
        OPEN, APPLY
    }
}

File: frontend/src/my_first_app/lib/models/course.dart
class Course {
  final int id;
  final String title;
  final String description;
  final String teacherName;
  final String status;
  final String permission;
  final String visibility;
  final String coverImage;
  bool isJoined;
  bool isPendingApproval;

  Course({
    required this.id,
    required this.title,
    required this.description,
    required this.teacherName,
    required this.status,
    required this.permission,
    this.visibility = 'PUBLIC',
    this.coverImage = '',
    this.isJoined = false,
    this.isPendingApproval = false,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: json['id'] as int? ?? 0,
      title: json['title'] as String? ?? '',
      description: json['description'] as String? ?? '',
      teacherName: (json['teacher_name'] ?? json['teachername']) as String? ?? '未知讲师',
      status: json['status'] as String? ?? 'IN_PROGRESS',
      permission: json['permission'] as String? ?? 'OPEN',
      visibility: json['visibility'] as String? ?? 'PUBLIC',
      coverImage: (json['cover_image'] ?? json['coverimage']) as String? ?? '',
      isJoined: (json['is_joined'] ?? json['isjoined'] as bool?) ?? false,
      isPendingApproval: false,
    );
  }
}

